eng/scripts/Automation-Sdk-UpdateChangelog.ps1

```diff
@@ -0,0 +1,389 @@
+        throw "New JAR file not found at: $NewJarPath"
+    }
+    if ((Get-Item $NewJarPath).PSIsContainer) {
+        throw "New JAR path is a directory, not a file: $NewJarPath"
+    }
+    
+    Write-Host "Running changelog generation tool..."
+    
+    # Determine the correct Maven command based on OS
+    $mvnCmd = if ($IsWindows -or $env:OS -match "Windows") { "mvn.cmd" } else { "mvn" }
+    
+    # Try to find Maven in PATH
+    $mvnPath = Get-Command $mvnCmd -ErrorAction SilentlyContinue
+    if (-not $mvnPath) {
+        throw "Maven executable '$mvnCmd' not found in PATH. Please ensure Maven is installed and available in your PATH."
+    }
+    
+    # Build the command as a single string for execution
+    $mvnCommand = "& `"$($mvnPath.Source)`" --no-transfer-progress clean package exec:java -q -f `"$changelogPomPath`" `"-DOLD_JAR=$OldJarPath`" `"-DNEW_JAR=$NewJarPath`""
+
+    # Output the full command for debugging
+    Write-Host ""
+    Write-Host "Changelog Generation Command:" -ForegroundColor Cyan
+    Write-Host $mvnCommand -ForegroundColor Yellow
+    Write-Host ""
+
+    # Execute the Maven command and capture output
+    try {
+        $output = Invoke-Expression $mvnCommand 2>&1
+        $stdout = ($output | Where-Object { $_ -is [string] }) -join "`n"
+        $stderr = ($output | Where-Object { $_ -is [System.Management.Automation.ErrorRecord] } | ForEach-Object { $_.Exception.Message }) -join "`n"
+        $exitCode = $LASTEXITCODE
+    }
+    catch {
+        throw "Failed to execute Maven command: $_"
+    }
+
+    
+    if ($exitCode -ne 0) {
+        Write-Host ""
+        Write-Host "Changelog generation failed with exit code $exitCode" -ForegroundColor Red
+        Write-Host ""
+        Write-Host "Standard Output:" -ForegroundColor Yellow
+        Write-Host $stdout
+        Write-Host ""
+        Write-Host "Error Output:" -ForegroundColor Red
+        Write-Host $stderr
+        throw "Changelog generation failed"
+    }
+    
+    # Parse the JSON output
+    try {
+        $changelogJson = $stdout | ConvertFrom-Json
+        return $changelogJson
+    }
+    catch {
+        Write-Error "Failed to parse changelog JSON output"
+        Write-Error "Output: $stdout"
+        throw
+    }
+}
+
+# Function to update the CHANGELOG.md file
+function Update-ChangelogFile {
+    param(
+        [string]$ChangelogPath,
+        [string]$NewChangelogText
+    )
+    
+    if (-not (Test-Path $ChangelogPath)) {
+        throw "CHANGELOG.md not found at: $ChangelogPath"
+    }
+
+    # output changelog text
+    Write-Host "-----------------NewChangelogText-----------------------"
+    Write-Host $NewChangelogText
+    Write-Host "----------------------------------------"
+
+    $oldChangelog = Get-Content $ChangelogPath -Raw
+    $lines = $oldChangelog -split "`n"
+    $newLines = @()
+    $newChangelogLines = $NewChangelogText -split "`n"
+    
+    $foundFirstVersion = $false
```

Comment:
Don't we want to ensure we are updating the section with the correct version instead of just the first? Perhaps you should read that from the pom file as well and find the matching version or use our existing changlog parsing to get that entry and update it.

Reply:
Updated in https://github.com/Azure/azure-sdk-for-java/commit/34011a3b60679234a1176e9c08b1f485df66faab and tested sccessfully